solvebio/cli/data.py

```diff
@@ -2,111 +2,125 @@
+        if not new_obj.is_folder:
+            raise SolveError('Object type {} already exists at location: {}'
+                             .format(new_obj.object_type, full_path))
     except NotFoundError:
-        base_remote_path, path_dict = \
-            Object.validate_full_path(base_remote_path)
-
+        # Create the folder
         if path_dict['path'] == '/':
             parent_object_id = None
         else:
-            obj = Object.get_by_full_path(base_remote_path,
-                                          assert_type='folder')
-            parent_object_id = obj.id
+            parent = Object.get_by_full_path(path_dict['parent_full_path'],
+                                             assert_type='folder')
+            parent_object_id = parent.id
 
-        # Create base folder
-        new_folder = Object.create(
+        # Make the API call
+        new_obj = Object.create(
             vault_id=vault.id,
             parent_object_id=parent_object_id,
             object_type='folder',
-            filename=local_start
+            filename=folder_name,
+            tags=tags or []
         )
 
-        print('Notice: Folder created for {0} at {1}'.format(
-            base_local_path,
-            new_folder.path,
-        ))
+        print('Notice: Folder created for {0} at {1}'
+              .format(folder_name, new_obj.path))
 
-    for root, dirs, files in os.walk(base_local_path):
+    return new_obj
 
-        # Create the sub-folders that do not exist on the remote
-        for d in dirs:
-            dirpath = os.path.join(
-                base_remote_path,
-                re.sub('^' + os.path.dirname(base_local_path), '', root).lstrip('/'),  # noqa
-                d
-            )
 
-            try:
-                Object.get_by_full_path(dirpath, object_type='folder')
-            except NotFoundError:
-                # Create the folder
-                if os.path.dirname(dirpath.split(':')[-1]) == '/':
-                    parent_object_id = None
-                else:
-                    parent_full_path = os.path.dirname(dirpath)
-                    parent = Object.get_by_full_path(
-                        parent_full_path, assert_type='folder')
-                    parent_object_id = parent.id
-
-                # Make the API call
-                new_obj = Object.create(
-                    vault_id=vault.id,
-                    parent_object_id=parent_object_id,
-                    object_type='folder',
-                    filename=d,
-                )
-
-                print('Notice: Folder created for {0} at {1}'
-                      .format(os.path.join(root, d), new_obj.path))
+def should_exclude(path, exclude_paths):
+    if not exclude_paths:
+        return False
+
+    if path in exclude_paths:
+        print("WARNING: Excluding path {} (via --exclude)".format(path))
+        return True
+
+    for exclude_path in exclude_paths:
+        if path.startswith(exclude_path):
```

Comment:
If this was changed to use fnmatch would it make things more complicated? `if fnmatch(exclude_path, path) return True`. Checking the prefix is just like always adding a `*` at the end of the exclude path.

Reply:
switched to using `fnmatch`